pysat/_instrument.py

```diff
@@ -45,21 +45,24 @@ class Instrument(object):
     orbit_info : dict
         Orbit information, {'index':index, 'kind':kind, 'period':period}.
         See pysat.Orbits for more information.
-    inst_module : module, optional
-        Provide instrument module directly.
-        Takes precedence over platform/name.
-    update_files : boolean, optional
+    inst_module : module
+        Provide instrument module directly, takes precedence over platform/name
+        (default=None)
+    update_files : boolean
         If True, immediately query filesystem for instrument files and store.
-    temporary_file_list : boolean, optional
+        Otherwise, assume nothing has changed from the last time files were
+        loaded (default=False)
+    temporary_file_list : boolean
         If true, the list of Instrument files will not be written to disk.
         Prevents a race condition when running multiple pysat processes.
-    strict_time_flag : boolean, option (True)
+        (default=False)
+    strict_time_flag : boolean
         If true, pysat will check data to ensure times are unique and
-        monotonically increasing.
-    multi_file_day : boolean, optional
+        monotonically increasing. (default=True)
+    multi_file_day : boolean or NoneType
         Set to True if Instrument data files for a day are spread across
         multiple files and data for day n could be found in a file
-        with a timestamp of day n-1 or n+1.
+        with a timestamp of day n-1 or n+1.  (default=None)
```

Comment:
So, here's a question.  Why is the default not just False?  The way you describe it seems like we just make the code more complicated and costly to run.  However, I finally figured out why you use None for an empty list so can be convinced otherwise.

Reply:
It started from the first python conference I was at that suggested using None in the method signatures for protection against unintended consequences. Particularly for items that can be modified in memory. Comes with the downside of kinda killing the transparency of the default value feature.

For this particular case, the None is used to distinguish between the user leaving things as default or specifying a False value. If left as default, then multi_file_day is set to False unless the instrument method provides its own value. A user supplied value overrides those.

Primarily though I use a None whenever a default value is gong to be an empty list or string.